test/protocol/produce_test.exs

```diff
@@ -10,7 +10,7 @@ defmodule KafkaEx.Protocol.Produce.Test do
       ]
     })
 
-    assert expected_request == request
+    assert expected_request == :erlang.iolist_to_binary(request)
```

Comment:
The `expected_request` should be changed to a list to match the expected output of the function.